services/lambda-pr-status-labeler/pr_status_bot/PRStatusBot.py

```diff
@@ -0,0 +1,355 @@
+
+    def _get_github_object(self):
+        """
+        This method returns github object initialized with Github personal access token
+        """
+        github_obj = Github(self.github_personal_access_token)
+        return github_obj
+
+    def _get_pull_request_object(self, github_obj, pr_number):
+        """
+        This method returns a PullRequest object based on the PR number
+        :param github_obj
+        :param pr_number
+        """
+        repo = github_obj.get_repo(self.repo)
+        pr_obj = repo.get_pull(int(pr_number))
+        return pr_obj
+
+    def _get_commit_object(self, github_obj, commit_sha):
+        """
+        This method returns a Commit object based on the SHA of the commit
+        :param github_obj
+        :param commit_sha
+        """
+        repo = github_obj.get_repo(self.repo)
+        commit_obj = repo.get_commit(commit_sha)
+        return commit_obj
+
+    def _is_mxnet_committer(self, github_obj, reviewer):
+        """
+        This method checks if the Pull Request reviewer is a member of MXNet committers
+        It uses the Github API for fetching team members of a repo
+        Only a Committer can access [read/write] to Apache MXNet Committer team on Github
+        Retrieved the Team ID of the Apache MXNet Committer team on Github using a Committer's credentials
+        """
+        team = github_obj.get_organization('apache').get_team(2413476)
+        return team.has_in_members(reviewer)
+
+    def _drop_other_pr_labels(self, pr, desired_label):
+        labels = pr.get_labels()
+        if not labels:
+            logging.info('No labels found')
+            return
+
+        for label in labels:
+            logging.info(f'Label:{label}')
+            if label.name.startswith('pr-') and label.name != desired_label:
+                try:
+                    logging.info(f'Removing {label}')
+                    # pr.remove_from_labels(label)
+                except Exception:
+                    logging.error(f'Error while removing the label {label}')
+
+    def _add_label(self, pr, label):
+        # drop other PR labels
+        self._drop_other_pr_labels(pr, label)
+
+        # check if the PR already has the desired label
+        if(self._has_desired_label(pr, label)):
+            logging.info(f'PR {pr.number} already contains the label {label}')
+            return
+
+        logging.info(f'BOT Labels: {label}')
+        # try:
+        #     pr.add_to_labels(label)
+        # except Exception:
+        #     logging.error(f'Unable to add label {label}')
+
+        # verify that label has been correctly added
+        # if(self._has_desired_label(pr, label)):
+        #     logging.info(f'Successfully labeled {label} for PR-{pr.number}')
+        return
+
+    def _has_desired_label(self, pr, desired_label):
+        """
+        This method returns True if desired label found in PR labels
+        """
+        labels = pr.get_labels()
```

Comment:
`return desired_label in set(pr.get_labels())`

Reply:
labels is a paginatedlist of Label objects [not strings]. Hence `set(pr.get_labels())` gives
```
TypeError: unhashable type: 'Label'
```

Reply:
Hence I have to iterate over the list & retrieve the object attribute